stdlib/public/core/StringGuts.swift

```diff
@@ -23,38 +23,92 @@ import SwiftShims
 @_fixed_layout
 public // FIXME
 struct _StringGuts {
-  // TODO 32-bit: Insert padding between fields
   public // FIXME for testing only
-  var _storage: (_StringObject, UInt)
```

Comment:
We would need a commitment on struct layout as part of ABI stability. The reason I went with tuples is that they are far more likely to be laid out ala C style. I don't know the current status. I'm ok with this for now.